### PR TITLE
WebIDL: Fix class string of null-prototype instance test

### PR DESCRIPTION
**Pull request by shvaikalesh**
Follow-up of #23140.
Spec: heycam/webidl#357.

_cc_ @domenic

### CODE REVIEW

File: WebIDL/ecmascript-binding/class-string-iterator-prototype-object.any.js [hunk 1]
```diff
@@ -31,16 +31,13 @@ test(t => {
...
-  t.add_cleanup(() => {
-    Object.setPrototypeOf(iteratorProto, proto);
-  });
-
-  Object.setPrototypeOf(iteratorProto, null);
+test(() => {
+  const iterator = (new URLSearchParams()).keys();
+  assert_equals(Object.prototype.toString.call(iterator), "[object URLSearchParams Iterator]");
 
-  assert_equals(Object.prototype.toString.call(iteratorProto), "[object Object]");
```
Comment (shvaikalesh) [inline WebIDL/ecmascript-binding/class-string-iterator-prototype-object.any.js:42]: `iteratorProto` has own `@@toStringTag`, so this assertion seems incorrect.
Chromium fails it: https://wpt.fyi/results/WebIDL/ecmascript-binding/class-string-iterator-prototype-object.any.html

File: WebIDL/ecmascript-binding/class-string-iterator-prototype-object.any.js [hunk 2]
```diff
@@ -31,16 +31,13 @@ test(t => {
...
-test(t => {
-  const proto = Object.getPrototypeOf(iteratorProto);
-  t.add_cleanup(() => {
-    Object.setPrototypeOf(iteratorProto, proto);
-  });
-
-  Object.setPrototypeOf(iteratorProto, null);
+test(() => {
+  const iterator = (new URLSearchParams()).keys();
+  assert_equals(Object.prototype.toString.call(iterator), "[object URLSearchParams Iterator]");
```
Comment (shvaikalesh) [inline WebIDL/ecmascript-binding/class-string-iterator-prototype-object.any.js:36]: To prevent caching.

**Review by domenic — COMMENTED** [review]
Thanks! However, perhaps we could split this into two tests? One that nulls out `iteratorProto`'s [[Prototype]], as before, but with a corrected `assert_equals`; and one that nulls out the instance's [[Prototype]]? I'm a bit sad about losing the coverage of the former case.

**Review by domenic — APPROVED** [review]

**Review by yuki3 — APPROVED** [review]
Defer details to Domenic.  Looks good to me.

**Pull request merged**